File: docs/guides/examples/py/src/buckets_change.py
from reduct import Client, BucketSettings, QuotaType

# Create a client with the base URL and API token
client = Client("http://localhost:8383", api_token="my-token")


async def change_bucket():
    # Get an existing bucket
    bucket = await client.get_bucket("existing-bucket")

    # Change the quota size of the bucket to 5GB
    new_settings = BucketSettings(
        quota_size=5000_000_000
    )
    await bucket.set_settings(new_settings)

    assert (await bucket.get_settings()).quota_size == 5000_000_000


if __name__ == "__main__":
    import asyncio

    asyncio.run(change_bucket())

File: docs/guides/examples/js/src/buckets_change.mjs
import {Client, QuotaType} from "reduct-js";
import assert from "node:assert";

// Create a new client with the server URL and an API token
const client = new Client("http://127.0.0.1:8383", {apiToken: "my-token"});

// Get an existing bucket
const bucket = await client.getBucket("example-bucket");

// Change the quota size of the bucket to 5GB
let new_settings = {
  quotaSize: 5000_000_000n,
}
await bucket.setSettings(new_settings);

assert((await bucket.getSettings()).quotaSize === 5000_000_000n);